### ISSUE TITLE
[Question] How to resume training with a saved checkpoint file?

### ISSUE THREAD

**Issue by sezaru** [BrikerMan/Kashgari#117]
Hello, sorry if this is a stupid question, but how can I resume a training I was doing with the latest checkpoint file available?

To add more context, I have this callback for my model:
```
filepath = "saved-model-{epoch:02d}-{val_acc:.2f}.hdf5"
checkpoint_callback = \
    ModelCheckpoint(path + filepath, monitor = 'acc', verbose = 1,
                    save_best_only = False, save_weights_only = False,
                    mode = 'auto', period=1)
```

And I'm training my model with this command:
```
embedding = BERTEmbedding('bert-base-multilingual-cased', sequence_length = 512)
hyper_params = CNNLSTMModel.__base_hyper_parameters__.copy()
model = CNNLSTMModel(embedding, hyper_parameters = hyper_params)

model.fit(tokenized_train_x.to_list(), train_y.to_list(),
          tokenized_val_x.to_list(), val_y.to_list(),
          epochs = 5,
          batch_size = 128,
          class_weight = True,
          fit_kwargs = {'callbacks': [tf_board_callback, checkpoint_callback]})
```

When the first epoch finished I got a file called `saved-model-01-1.00.hdf5`. Now what I want is to load that file and then continue training from that (load that first epoch model and continue from second epoch).

How can I do that? I tried the following code but I'm not sure if I'm doing it right at all:
```
embedding = BERTEmbedding('bert-base-multilingual-cased', sequence_length = 512)
hyper_params = CNNLSTMModel.__base_hyper_parameters__.copy()
model = CNNLSTMModel(embedding, hyper_parameters = hyper_params)

model.build_model(tokenized_train_x.to_list(), train_y.to_list(),
          tokenized_val_x.to_list(), val_y.to_list())
model.model.load_weights(path + 'saved-model-01-1.00.hdf5')

model.fit(tokenized_train_x.to_list(), train_y.to_list(),
          tokenized_val_x.to_list(), val_y.to_list(),
          epochs = 4,
          batch_size = 128,
          class_weight = True,
          fit_kwargs = {'callbacks': [tf_board_callback, checkpoint_callback]})
```

Doing that it seems like I'm getting the correct loss and acc from previous epoch as far as I can tell, but I'm not sure if everything is loaded correctly and I had to change `epochs` from 5 to 4 to compensate the first epoch already trained (it would be nicer to somehow say to it that I'm already in second epoch, that way my checkpoint `saved-model-01-1.00.hdf5` would not be overridden by the second epoch).

Thanks in advance!

**Comment by BrikerMan** [comment]
You are doing it right, but, for tf.keras, even if you are started from a  saved model, it always starts from the epoch-0. Here is an example:

```python
model = CNNLSTMModel()
model.fit(x, y) # start from epoch 0
model.fit(x, y) # Also starts from epoch 0
```

So, I recommend changing your code like this.

```python
filepath = "saved-model-{epoch:02d}-{val_acc:.2f}.hdf5"
# Add a run sub-dir for every run
path = path + '/run_0/'
checkpoint_callback = \
    ModelCheckpoint(path + filepath, monitor = 'acc', verbose = 1,
                    save_best_only = False, save_weights_only = False,
                    mode = 'auto', period=1)
```

**Comment by sezaru** [comment]
Thanks @BrikerMan!

I was looking at the tf.keras documentation and it seems that it model `fit` function has a parameter called `initial_epoch`. If I understood it correctly I can change it to any other number and the epoch will start from there, I will try it later and if it works I will update it here.

**Comment by BrikerMan** [comment]
@sezaru Any update?

**Comment by sezaru** [comment]
Hey @BrikerMan 
Sorry I took so long to reply...

So, it works great! Basically the only difference to the code I pasted here before is the `model.fit` call which now includes the `initial_epoch` variable.

In my case I already have a checkpoint for the epoch 1, so I need to set `initial_epoch` to 1 and it will start `fit` directly from epoch 2:

```
model.fit(tokenized_train_x.to_list(), train_y.to_list(),
          tokenized_val_x.to_list(), val_y.to_list(),
          epochs = 5,
          batch_size = 128,
          class_weight = True,
          fit_kwargs = {'callbacks': [tf_board_callback, checkpoint_callback], 'initial_epoch': 1})
```

```
Epoch 2/5
  49/7317 [..............................] - ETA: 8:17:27 - loss: 1.1702 - acc: 0.7894
```

I guess it would be nice to have something mentioning it in the Kashgari docs talking about the checkpoint callback.

Thanks!

**Comment by BrikerMan** [comment]
Will add to the document~
